x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/ChoroplethMap/index.tsx

```diff
@@ -263,16 +247,21 @@ export const ChoroplethMap: React.SFC<Props> = props => {
     }
   }, [map, tooltipState]);
 
+  const toolTipProps = { name: '', value: 0, docCount: 0, ...tooltipState };
+  const hasMinMax = Number.isFinite(min) && Number.isFinite(max);
+
   // render map container and tooltip in a hidden container
   return (
     <div>
       <div ref={containerRef} style={{ height: 256 }} />
       <div style={{ display: 'none' }}>
         <div ref={popupContainerRef}>
-          <ChoroplethToolTip {...tooltipState} />
+          <ChoroplethToolTip {...toolTipProps} />
         </div>
       </div>
-      <Legend getColorStyle={getProgressionColor} min={min} max={max} />
+      {hasMinMax ? (
+        <Legend getColorStyle={getProgressionColor} min={min} max={max} />
+      ) : null}
```

Comment:
I think what we really want is to say: hide the legend if there are no items. `min` and `max` should be defined if there are items (if not we should clean up the list before passing it to `ChoroplethMap` imo)

```js
      {items.length > 0 && (
        <Legend getColorStyle={getProgressionColor} min={min} max={max} />
      )}
```